ArchiSteamFarm/Plugins/Interfaces/IGitHubPluginUpdates.cs

```diff
@@ -0,0 +1,180 @@
+	/// <param name="releaseAssets">Available release assets for auto-update. Those come directly from your release on GitHub.</param>
+	/// <remarks>
+	///     Default implementation will select release asset in following order:
+	///     - {PluginName}-V{Major}-{Minor}-{Build}-{Revision}.zip
+	///     - {PluginName}-V{Major}-{Minor}-{Build}.zip
+	///     - {PluginName}-V{Major}-{Minor}.zip
+	///     - {PluginName}-V{Major}.zip
+	///     - {PluginName}.zip
+	///     - *.zip, if exactly 1 release asset matching in the release
+	///     Where:
+	///     - {PluginName} is <see cref="IPlugin.Name" />
+	///     - {Major} is target major ASF version (A from A.B.C.D)
+	///     - {Minor} is target minor ASF version (B from A.B.C.D)
+	///     - {Build} is target build (patch) ASF version (C from A.B.C.D)
+	///     - {Revision} is target revision ASF version (D from A.B.C.D)
+	///     - * is a wildcard matching any string value
+	///     For example, when updating MyAwesomePlugin with ASF version V6.0.1.3, it will select first zip file from available ones in the following order:
+	///     - MyAwesomePlugin-V6.0.1.3.zip
+	///     - MyAwesomePlugin-V6.0.1.zip
+	///     - MyAwesomePlugin-V6.0.zip
+	///     - MyAwesomePlugin-V6.zip
+	///     - MyAwesomePlugin.zip
+	///     - *.zip
+	/// </remarks>
+	/// <returns>Target release asset from those provided that should be used for auto-update. You may return null if the update is unavailable, for example, because ASF version/variant is determined unsupported, or due to any other reason.</returns>
+	Task<ReleaseAsset?> GetTargetReleaseAsset(Version asfVersion, string asfVariant, Version newPluginVersion, IReadOnlyCollection<ReleaseAsset> releaseAssets) {
+		ArgumentNullException.ThrowIfNull(asfVersion);
+		ArgumentException.ThrowIfNullOrEmpty(asfVariant);
+		ArgumentNullException.ThrowIfNull(newPluginVersion);
+
+		if ((releaseAssets == null) || (releaseAssets.Count == 0)) {
+			throw new ArgumentNullException(nameof(releaseAssets));
+		}
+
+		Dictionary<string, ReleaseAsset> assetsByName = releaseAssets.ToDictionary(static asset => asset.Name, StringComparer.OrdinalIgnoreCase);
+
+		List<string> matches = [
+			$"{Name}-V{asfVersion.Major}-{asfVersion.Minor}-{asfVersion.Build}-{asfVersion.Revision}.zip",
+			$"{Name}-V{asfVersion.Major}-{asfVersion.Minor}-{asfVersion.Build}.zip",
+			$"{Name}-V{asfVersion.Major}-{asfVersion.Minor}.zip",
+			$"{Name}-V{asfVersion.Major}.zip",
+			$"{Name}.zip"
+		];
+
+		foreach (string match in matches) {
+			if (assetsByName.TryGetValue(match, out ReleaseAsset? targetAsset)) {
+				return Task.FromResult<ReleaseAsset?>(targetAsset);
+			}
+		}
+
+		// The very last fallback in case user uses different naming scheme
+		HashSet<ReleaseAsset> zipAssets = releaseAssets.Where(static asset => asset.Name.EndsWith(".zip", StringComparison.OrdinalIgnoreCase)).ToHashSet();
+
+		return Task.FromResult(zipAssets.Count == 1 ? zipAssets.First() : null);
```

Comment:
Should we provide any kind of error to the user if we can’t decide on the asset to use?